Automattic/wp-calypso
# Jetpack Settings: Add Theme Enhancements card under Writing Settings for Jetpack sites

### Introduction

This PR is part of #9171. It introduces a Theme Enhancements card under Writing Settings. This card will be used to manage the settings of the Infinite Scroll and Mobile Theme Jetpack modules.

### Preview

#### Both modules activated, some settings are disabled
![](https://cldup.com/xCIIPi8iRc.png)

#### Initial loading
![](https://cldup.com/knDrQ3NSGK.png)

#### Loading while saving
![](https://cldup.com/vS2p_mDNWX.png)

#### Disabled modules (settings are hidden)
![](https://cldup.com/lBqsNkjADP.png)

### Testing

* Checkout this branch.
* Select one of your Jetpack sites with Jetpack 4.5 or newer.
* Go to `/settings/writing/$site`, where `$site` is the slug of your Jetpack site.
* Verify the Theme Enhancements card is displayed properly.
* Play with the main toggles that enable the modules, and verify they save correctly in your Jetpack site (note that notices are implemented separately in #10583).
* Verify that the settings of a module are displayed only when the module is active.
* Play around with the settings of each module and click "Save Settings". Verify they save properly in your Jetpack site.
* Verify the links of the info popovers are working as expected and lead to the right place.
* Test with a Jetpack site that has Jetpack 4.4.2 or older, and verify the card is not shown.
* Test with a WordPress.com site and verify that the card is not shown.
